sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeTest.java

```diff
@@ -448,6 +448,21 @@ public void doNotShowUpdateSetupFailedDialogInBackground() {
         SharedPreferencesManager.remove(PREFERENCE_KEY_UPDATE_SETUP_FAILED_MESSAGE_KEY);
     }
 
+    @Test
+    public void doNotShowUpdateSetupFailedDialogInNullForegroundActivity() {
+        when(SharedPreferencesManager.getString(PREFERENCE_KEY_UPDATE_SETUP_FAILED_MESSAGE_KEY)).thenReturn("failed_message");
+
+        /* Trigger call. */
+        Distribute.getInstance().showUpdateSetupFailedDialog();
+        mActivity = null;
+        start();
+
+        /* Verify dialog. */
+        verify(mDialogBuilder, never()).create();
+        verifyStatic(SharedPreferencesManager.class, never());
```

Comment:
We don't need to check `SharedPreferencesManager`. All we need to verify is that the call doesn't throw NPE and it doesn't reach the dialog creation. So I think checking that dialog builder was not created is enough here.